service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -155,4 +158,77 @@
         }
         return result;
     }
+
+    @Override
+    public List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId,
+                                                    List<String> genericAssayStableIds, String projection)
+        throws MolecularProfileNotFoundException {
+        
+        validateMolecularProfile(molecularProfileId);
+        List<String> sampleIds = sampleListRepository.getAllSampleIdsInSampleList(sampleListId);
+        if (sampleIds.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return fetchGenericAssayData(molecularProfileId, sampleIds, genericAssayStableIds, projection);
+    }
```

Comment:
I smell code duplication here but I am not able to spot this. In essence I expect that a call to `getGenericAssayData` would call method `getGenericAssayDataInMultipleMolecularProfiles` like so:
```
public List<GenericAssayData> getGenericAssayData(String molecularProfileId, String sampleListId,
                                                    List<String> genericAssayStableIds, String projection)
        throws MolecularProfileNotFoundException {
       String[] molecularProfileIds = { molecularProfileId };
       return getGenericAssayDataInMultipleMolecularProfiles(molecularProfileIds, ....);
    }
```

Either that, or I would expect method `getGenericAssayDataInMultipleMolecularProfiles` to make use of the `fetchGenericAssayData`.

I have not dived into the code to deep, but glancing this code I would like to discuss whether this code may be optimized.

Reply:
Thanks for your comments! Just updated the solution, could you take a look again?